metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/TextProcessor.java

```diff
@@ -75,8 +80,13 @@ public ResourceExtractionResultImpl extractMetadata(Resource resource, String de
     // Create thumbnails in case of PDF file.
     final List<Thumbnail> thumbnails;
     if (PDF_MIME_TYPE.equals(detectedMimeType)) {
-      thumbnails = thumbnailGenerator.generateThumbnails(resource.getResourceUrl(),
-          detectedMimeType, resource.getContentFile()).getRight();
+      final Path pdfImage = pdfToImageConverter.convertToPdf(resource.getContentPath());
+      try {
+        thumbnails = thumbnailGenerator.generateThumbnails(resource.getResourceUrl(),
+                PNG_MIME_TYPE, pdfImage.toFile(), true).getRight();
+      } finally {
+        removePdfImageFileSilently(pdfImage);
```

Comment:
Perhaps reuse the method in the pdfToImageConverter?

Reply:
Fixed.